### PR TITLE
Provide support for scenarios. Fix #182

### LINKED ISSUE #182: Provide support for Wiremock scenarios
Support for Wiremock stateful behaviour as described in [Wiremock documentation](http://wiremock.org/stateful-behaviour.html) as it would allow UI testers to run complex test scenarios.

### PR DESCRIPTION
**Pull request by jkubrynski**

### CODE REVIEW

File: accurest-core/src/main/groovy/io/codearte/accurest/file/Contract.groovy [hunk 1]
```diff
@@ -8,10 +8,23 @@ import java.nio.file.Path
...
-	Contract(Path path, boolean ignored) {
+	Contract(Path path, boolean ignored, int groupSize, Integer order) {
+		this.groupSize = groupSize
 		this.path = path
 		this.ignored = ignored
+		this.order = order
 	}
 
+	@Override
+	public String toString() {
```
Comment (marcingrzejszczak) [inline accurest-core/src/main/groovy/io/codearte/accurest/file/Contract.groovy:22]: You could have annotated the class with `@ToString`

File: accurest-core/src/test/resources/directory/with/scenario/01_login.groovy [hunk 2]
```diff
@@ -0,0 +1 @@
+package directory.with.scenario
```
Comment (marcingrzejszczak) [inline accurest-core/src/test/resources/directory/with/scenario/01_login.groovy]: unnecessary?

File: accurest-core/src/test/resources/directory/with/scenario/02_showCart.groovy [hunk 3]
```diff
@@ -0,0 +1 @@
+package directory.with.scenario
```
Comment (marcingrzejszczak) [inline accurest-core/src/test/resources/directory/with/scenario/02_showCart.groovy]: unnecessary?

File: accurest-core/src/test/resources/directory/with/scenario/03_logout.groovy [hunk 4]
```diff
@@ -0,0 +1 @@
+package directory.with.scenario
```
Comment (marcingrzejszczak) [inline accurest-core/src/test/resources/directory/with/scenario/03_logout.groovy]: unnecessary?

File: accurest-core/src/main/groovy/io/codearte/accurest/file/Contract.groovy (hunk 1 above)
Comment (jkubrynski) [inline accurest-core/src/main/groovy/io/codearte/accurest/file/Contract.groovy:22]: I don't want to dump full path - just a fileName

File: accurest-core/src/test/resources/directory/with/scenario/01_login.groovy (hunk 2 above)
Comment (jkubrynski) [inline accurest-core/src/test/resources/directory/with/scenario/01_login.groovy]: Yep - looks like added by IDEA during refactoring

**Comment by marcingrzejszczak** [comment]
So the idea is that you take the `01_` prefixed groovy file. Its state is `Started`. Then each subsequent number-prefixed file will have a required state `StateX` where X is the number in the prefix. It will have the next state as the next filename if that one exists. Am I correct?

**Comment by jkubrynski** [comment]
Yep - that was my idea. Scenario is considered only inside single directory

File: accurest-core/src/main/groovy/io/codearte/accurest/file/Contract.groovy (hunk 1 above)
Comment (szpak) [inline accurest-core/src/main/groovy/io/codearte/accurest/file/Contract.groovy:22]: `@ToString(includePackage = false)`?

Comment (jkubrynski) [inline accurest-core/src/main/groovy/io/codearte/accurest/file/Contract.groovy:22]: @szpak Will it affect path field?

Comment (szpak) [inline accurest-core/src/main/groovy/io/codearte/accurest/file/Contract.groovy:22]: I thought you were writing about full package path for a class :).

**Comment by marcingrzejszczak** [comment]
LGTM

**Pull request merged**